### PR TITLE
subset(): test if one range is a subset of another

### PR DESCRIPTION
**Pull request by isaacs**
cc @ljharb

### CODE REVIEW

File: test/ranges/subset.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const t = require('tap')
+const subset = require('../../ranges/subset.js')
+
+// sub, dom, expect, [options]
+const cases = [
+  ['1.2.3', '1.2.3', true],
+  ['1.2.3 1.2.4', '1.2.3', true],
```
Comment (ljharb) [inline test/ranges/subset.js:7]: this seems off - the first one allows 1.2.4, but the second one doesn’t, so shouldn’t it be false?

File: test/ranges/subset.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  ['1.2.3 2.3.4 || 2.3.4', '3', false],
+  ['^1.2.3-pre.0', '1.x', false],
+  ['^1.2.3-pre.0', '1.x', true, { includePrerelease: true }],
+  ['>2 <1', '3', true],
+  ['1 || 2 || 3', '>=1.0.0', true],
+  ['*', '*', true],
+  ['*', '>=0.0.0-0', false], // XXX technically wrong, should detect this?
+  ['>=1.0.0 <=1.0.0 || 2.0.0', '1.0.0 || 2.0.0', true],
+  ['>=1.0.0 <=1.0.0 || 2.0.0', '1.0.0 || 2.0.0', true],
```
Comment (ljharb) [inline test/ranges/subset.js:16]: this is a repeat of the previous line

File: test/ranges/subset.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  ['1.2.3', '1.2.3', true],
+  ['1.2.3 1.2.4', '1.2.3', true],
+  ['1.2.3 2.3.4 || 2.3.4', '3', false],
+  ['^1.2.3-pre.0', '1.x', false],
+  ['^1.2.3-pre.0', '1.x', true, { includePrerelease: true }],
+  ['>2 <1', '3', true],
+  ['1 || 2 || 3', '>=1.0.0', true],
+  ['*', '*', true],
+  ['*', '>=0.0.0-0', false], // XXX technically wrong, should detect this?
+  ['>=1.0.0 <=1.0.0 || 2.0.0', '1.0.0 || 2.0.0', true],
```
Comment (ljharb) [inline test/ranges/subset.js:15]: does the former not allow 3, while the latter does not, hence false?

File: test/ranges/subset.js (hunk 1 above)
Comment (isaacs) [reply test/ranges/subset.js:7]: The first one is a null set.  It's `1.2.3` AND `1.2.4`, not `1.2.3 || 1.2.4`.

File: test/ranges/subset.js (hunk 2 above)
Comment (isaacs) [reply test/ranges/subset.js:16]: yep.  fixing.

File: test/ranges/subset.js (hunk 3 above)
Comment (isaacs) [reply test/ranges/subset.js:15]: The former does not allow 3.  It is:

- versions that are `>=1.0.0` AND `<=1.0.0` (ie, only 1.0.0!), OR
- versions that are 2.0.0.

So the sets are equivalent.

File: test/ranges/subset.js (hunk 1 above)
Comment (ljharb) [reply test/ranges/subset.js:7]: gotcha

File: test/ranges/subset.js (hunk 3 above)
Comment (ljharb) [reply test/ranges/subset.js:15]: ah, the above means the former is "1.0.0 || 2.0.0", so true is correct :-)

**Pull request merged**